### ISSUE TITLE
update inside / outside

### ISSUE THREAD

**Issue by pokey** [pokey/cursorless-vscode#254]
- "inside" / "outside" / "pair" can be applied to anything that has an interior and scope starters / enders, eg an `if` statement, surrounding pairs, functions, strings, etc.  Note that they do *not* apply to items in delimited sequences, such as args
- separately, we have "leading" / "trailing" that refer to the delimiters before or after items in delimited sequences, eg the commas in an arg list, or whitespace for tokens
- We want "inside" without any scope type to automatically refer to the nearest applicable scope, not including the "space" surrounding pari type
- The above modifiers will just go into a `modifiers.csv` in cursorless-settings
- If you say "chuck inside list" it will remove everything inside list including whitespace padding
- If you say "clear inside list" it will not remove whitespace padding inside
- If you say "chuck leading inside list" it will remove leading padding in the list
- If you say "chuck trailing string" it will remove whitespace after the string
- If you say "chuck list" it will remove list and trailing whitespace

**Comment by AndreasArvidsson** [comment]
I'm gonna take a crack at inside element because that one is so useful

**Comment by pokey** [comment]
Yeah that's fair. Was hoping to wait until #629, but this one is too useful to hold off I think

Prob easiest to just add an `interiorRange` attribute to the legacy [`SelectionContext`](https://github.com/cursorless-dev/cursorless/blob/main/src/typings/Types.ts#L56)?

Once we have #629 we can add weak expansion support that will stop if it hits a scope type that supports inside, so that we can just say "take inside air" for anything inside an element.

**Comment by AndreasArvidsson** [comment]
Exactly the plan

**Comment by AndreasArvidsson** [comment]
Started work in #851
